text/stacks/0003-stack-descriptor.md

```diff
@@ -0,0 +1,85 @@
+# Stack Descriptor and Tooling
+
+## Summary
+
+The stack-building codebase,
+[`create-stack`](https://github.com/paketo-buildpacks/stacks/tree/main/create-stack)
+should be moved into the `jam` tooling CLI, and allow users to configure their
+stack build via a common configuration file.
+
+## Motivation
+
+Today, `create-stack` is highly coupled to the `bionic` stacks we currently
+support. This makes it difficult to support alternative stacks and platforms or
+to allow the codebase to be reused by external parties for developing their own
+stack.
+
+## Detailed Explanation
+
+The `create-stack` codebase will be refactored to allow users to build stacks
+using a configuration file with the following format:
+
+```toml
+id = "<stack id>"                # required
+homepage = "<homepage uri>"      # optional
+maintainer = "<maintainer name>" # optional
+
+platforms = ["linux/amd64", "linux/arm64", "<other-platform>"] # optional; default = ["linux/amd64"]; choices defined in https://docs.docker.com/desktop/multi-arch/
+
+[distro]
+  name = "<os distro name>"       # optional
+  version = "os distro version>"  # optional
+
+[build]
+  tag = "<image tag>"                 # required
+  dockerfile = "<path to dockerfile>" # required
+  description = "<image description>" # optional
+
+  uid = 1001                   # required
+  gid = 1000                   # required
+  shell = "<user login shell>" # optional; default = "/sbin/nologin"
+
+  [build.args] # optional
+    example-arg = "<example-value>"
+
+[run]
+  tag = "<image tag>"                 # required
+  dockerfile = "<path to dockerfile>" # required
+  description = "<image description>" # optional
+
+  uid = 1002                   # required
+  gid = 1000                   # required
+  shell = "<user login shell>" # optional; default = "/sbin/nologin"
+
+  [run.args] # optional
+    example-arg = "<example-value>"
+
+[deprecated]
+  legacy-sbom = true # optional; default = false; enables https://github.com/paketo-buildpacks/rfcs/blob/main/text/stacks/0001-stack-package-metadata.md
+  mixins = true      # optional; default = false; enables the inclusion of the `io.buildpacks.stack.mixins` label
+```
+
+Then the codebase will be merged into the `jam` CLI under a `create-stack`
+command. Once complete, a user should be able to build a stack with a given
+descriptor file, `stack.toml`, using the following command:
+
+`jam create-stack --config stack.toml`
+
+## Rationale and Alternatives
+
+We could abandon the `create-stack` codebase entirely and expect that anyone
```

Comment:
As someone with essentially no exposure to the stack creation, can you briefly describe the downside to the `Dockerfile` approach? I can infer/assume but I'd like to hear your thoughts.

Reply:
The biggest concern I have between the Dockerfile-only interface and what is proposed here is that we're co-mingling the parts that make up a "base image" (os-specific modifications to an image to make it a reasonable base to build on top of) and a "stack" (the parts of the image that are codified in the Buildpacks Platform Spec or in Paketo RFCs). The effort here is to try to separate those things so its a bit clearer where the line is.

Additionally, we have some features in the existing stacks that aren't part of the Buildpacks Platform Spec that are non-trivial to implement inline in a Dockerfile. Specifically, I am thinking of the code that generates our existing image label-based SBOM for run images. Ideally, I'd like to be able to move the existing stacks onto this new configuration format and tooling since we'll be maintaining them for the next year before they go out of support.